src/norm/postgres.nim

```diff
@@ -98,6 +98,14 @@ proc createTables*[T: Model](dbConn; obj: T) =
       fkGroups.add "FOREIGN KEY($#) REFERENCES $#($#)" %
         [obj.col(fld), typeof(get val.model).table, typeof(get val.model).col("id")]
 
+    when obj.dot(fld).hasCustomPragma(fk):
+      const pragVal = $(obj.dot(fld).getCustomPragmaVal(fk))
+      const fkTargetIsModel = isModel(none(typeof(obj.dot(fld).getCustomPragmaVal(fk))))
```

Comment:
Why are you constructing a `None` and checking that against `is Model` instead of just checking the type? `isModel` is for object checks, not type checks. In other words, `obj.dot(fld).getCustomPragmaVal(fk) is Model` would probably do the trick.

Reply:
Yeah,  that's a better way to handle it